jadx-gui/src/main/java/jadx/gui/utils/search/CodeIndex.java

```diff
@@ -27,15 +29,35 @@ public synchronized void removeForCls(JavaClass cls) {
 		values.removeIf(v -> v.getJavaNode().getTopParentClass().equals(cls));
 	}
 
-	private boolean isMatched(StringRef key, String str, boolean caseInsensitive) {
-		return key.indexOf(str, caseInsensitive) != -1;
+	private boolean isMatched(StringRef key, String str, boolean caseInsensitive, boolean useRegex) {
+		if (caseInsensitive && useRegex) {
+			try {
+				Pattern pattern = Pattern.compile(str, Pattern.CASE_INSENSITIVE);
```

Comment:
`isMatched` is called for each `CodeNode`, hence it is called several thousand times . For efficiency it would be good if you could manage to compile the regex pattern only once for a complete search. 

May be it is time to switch from search method(s) that get the search options as parameters to a search predicate that is created once and then applied to all items to be searched?

Reply:
Thanks for the quick response.  That is a good point about the efficiency issues with compiling in the isMatched function.  Ill make some changes to fix this.

Reply:
I just created a new class SearchSettings to store all of the options provided by the user and pass it as the input to the search methods.  I also created a SearchImpl class to check if a match was found.  I think this is a bit cleaner and will enable the Regex to be compiled prior to running the search.  It should also make it easier to add new options in the future should they be needed.  

Its worth pointing out that the SearchEngine API used for highlighting appears to compile Regex at search time as well so there might still be a hit to efficiency.  It may be possible extend the new search classes i made to perform the highlighting function which will enable the regex to be compiled ahead of time.  Let me know if that is a concern and I can look into updating it.

Reply:
Thanks. Your implementation is not exactly what I had in mind but the major point regarding the regex compilation is solved.